pkg/storage/segment/segment.go

```diff
@@ -144,6 +153,19 @@ func (sn *streeNode) get(st, et time.Time, cb func(sn *streeNode, d int, t time.
 	}
 }
 
+func (sn *streeNode) cleanup(timeThreshold time.Time, cb func(depth int, t time.Time)) {
+	rel := sn.cleanupRelationship(timeThreshold)
+	if sn.present && (rel == inside || rel == match) {
+		cb(sn.depth, sn.time)
```

Comment:
This node (`sn`) should also be deleted from the segment tree after we call `cb`